### ISSUE TITLE
Hotkey for Stop

### ISSUE THREAD

**Issue by Elenkhos** [aleab/toastify#88]
Could you add a hotkey to stop the music? It existed in the original version and it was extremely useful in some cases. Play/Pause doesn't work because it starts the music if it is paused and that's not what I want when executing some AutoHotkey scripts.

Or could you give a hint to right direction how to stop media (with AutoHotkey or some other way)? For some reason `Send, {Media_Stop}` doesn't work.

**Comment by aleab** [comment]
Sure, I can add it!

**Comment by aleab** [comment]
I'll release a version with the new hotkey soon™ when I've fixed a bunch of other issues.

Autohotkey's `Send {Media_Stop}` or `SendInput {Media_Stop}` work fine for me, except occasionally when some control in some window is focused or expecting input, I suppose.

**Comment by Elenkhos** [comment]
> Sure, I can add it!

Great, thanks!

> Autohotkey's Send {Media_Stop} or SendInput {Media_Stop} work fine for me, except occasionally when some control in some window is focused or expecting input, I suppose.

Well, that's interesting. I tried every send mode and Spotify indeed shows the desktop overlay with input and event modes but the music doesn't stop if Spotify is minimised to tray (or the window isn't active). In order to work Spotify window must be _active_.

I would of course prefer sending commands directly with AutoHotkey but for some reason there is this limitation that Spotify window must be open and active. Does it work for you if you minimise Spotify to tray?

**Comment by aleab** [comment]
It works fine for me when Spotify is minimized to the tray.
However, it doesn't work if I enable the *"Show desktop overlay when using media keys"* option in Spotify. Go figure... It's definitely a bug on Spotify's end.

**Comment by Elenkhos** [comment]
Yes, I tried disabling the desktop overlay and got the same behaviour. So this is Spotify's bug like you said. I [reported](https://community.spotify.com/t5/Desktop-Windows/Bug-Media-Stop-when-minimised-to-tray/td-p/4545335) this in Spotify community but my past experiences tell that it's higly unlikely they would ever fix this issue.

Well, now I have to choose between desktop overlay and stop command... Or use Toastify's stop hotkey as a workaround if it works with desktop overlay enabled.

Thanks @aleab for your insight and for the hotkey in advance!

**Comment by Elenkhos** [comment]
You have been extremely helpful @aleab! I really appreciate your efforts.

I have used the same code to get the right window id for Spotify. I have used it to automatically search for lyrics and read out loud the now playing info. And also to display a little notification with the same info since Toastify stopped working. Quick screenshot:

![image](https://user-images.githubusercontent.com/39801064/43995195-76323e8e-9db2-11e8-9f53-08995a342ce8.png)

But I didn't know that I could use `PostMessage` to send media commands directly to Spotify. I googled a bit and found all the other commands and I'm going to chance all of my media hotkeys to send the commands directly to Spotify (and other apps). This must have been one of the most valuable feature request I've ever made.

So you don't need to add the hotkey for stop to Toastify just for me. But I think some users might find it helpful, too.

**Comment by aleab** [comment]
Happy I could be of help! 😃

**Comment by Elenkhos** [comment]
I will post this if someone is interested. An alternative way to identify the right Spotify window:
```AutoHotkey
; Pause
<^!Down::
DetectHiddenWindows, On
SetTitleMatchMode, 2
PostMessage, 0x0319,, 0x002F0000,,  -  ahk_exe Spotify.exe ahk_class Chrome_WidgetWin_0 ; Right window has " - " between artist and song name.
; DetectHiddenWindows, Off ; These are thread specific
; SetTitleMatchMode, 1
Return
```
I'm not sure which one is more efficient 🤔

**Comment by Elenkhos** [comment]
Good improvement! In fact, my startup script starts playing music just before this main script is executed so I didn't notice (yet) that there would be this issue if the music isn't playing.  But with regex it's easily avoided. Thanks again for your sharp thinking!

**Comment by aleab** [comment]
Closing the issue as this has been implemented in [v1.10.10](https://github.com/aleab/toastify/releases/tag/v1.10.10).
